lib/utils/job-utils/catalog-searcher.js

```diff
@@ -79,100 +79,212 @@ function searchCatalogDataFactory(
             }
 
             return driveId.identifier in filter || driveId.devName in filter;
-        }).tap(function (driveIds) {
+        })
+        .tap(function (driveIds) {
+            // get virtualDisk data from megaraid-virtual-disks catalog
             var foundVdHasValue = _.find(driveIds, function (driveId) {
                 return !_.isEmpty(driveId.virtualDisk);
             });
 
             if (!foundVdHasValue) {
                 return;
             }
-            // get virtualDisk data from megaraid-virtual-disks catalog
             return waterline.catalogs.findMostRecent({
                 node: nodeId,
                 source: 'megaraid-virtual-disks'
-            }).then(function (virtualDiskCatalog) {
+            })
+            .then(function (virtualDiskCatalog) {
                 if (!_.has(virtualDiskCatalog, 'data.Controllers[0]')) {
                     return Promise.reject(
                         new Error('Could not find megaraid-virtual-disks catalog data.'));
                 }
-
                 virtualDiskData = virtualDiskCatalog.data;
-                return virtualDiskData.Controllers;
-            }).tap(function () {
-                // find controller data from megaraid-controllers catalog
-                return waterline.catalogs.findMostRecent({
-                    node: nodeId,
-                    source: 'megaraid-controllers'
-                }).then(function (controllerCatalog) {
-                    controllerData = _.get(controllerCatalog, 'data');
+            });
+        })
+        .tap(function () {
+            // get controller vendor information from megaraid-controllers
+            return waterline.catalogs.findMostRecent({
+                node: nodeId,
+                source: 'megaraid-controllers'
+            })
+            .then(function (controllerCatalog) {
```

Comment:
Since the `megaraid-controllers` catalogs may not exist, so you need a safely check before using it.